File: MyClub/MyClub.Services/Database/Entities/MatchTicket.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyClub.Services.Database
{
    public class MatchTicket
    {
        [Key]
        public int Id { get; set; }

        [Required(ErrorMessage = "Released quantity is required")]
        [Range(0, 100, ErrorMessage = "Released quantity must be between 0 and 100")]
        public int ReleasedQuantity { get; set; }

        [Required]
        [Column(TypeName = "decimal(10,2)")]
        public decimal Price { get; set; }

        // Match relationship
        public int MatchId { get; set; }

        [ForeignKey("MatchId")]
        public virtual Match Match { get; set; }

        // Stadium Sector relationship
        public int StadiumSectorId { get; set; }

        [ForeignKey("StadiumSectorId")]
        public virtual StadiumSector StadiumSector { get; set; }

        // Navigation collections
        public virtual ICollection<UserTicket> UserTickets { get; set; }

        public int UsedQuantity => ReleasedQuantity - AvailableQuantity;

        public int AvailableQuantity { get; set; }
    }
} 

File: MyClub/UI/myclub_desktop/lib/models/match_ticket.dart
import 'package:myclub_desktop/models/stadium_sector.dart';

class MatchTicket {
  int? id;
  int? matchId;
  StadiumSector? stadiumSector;
  int? releasedQuantity;
  double? price;
  int? availableQuantity;
  int? usedQuantity;

  MatchTicket({this.id, this.matchId, this.stadiumSector,
  this.releasedQuantity, this.availableQuantity, this.price, this.usedQuantity});

  factory MatchTicket.fromJson(Map<String, dynamic> json) {
    return MatchTicket(
      id: json['id'] is int ? json['id'] : int.tryParse(json['id']?.toString() ?? '0'),
      matchId: json['matchId'] is int ? json['matchId'] : int.tryParse(json['matchId']?.toString() ?? '0'),
      stadiumSector: json['stadiumSector'] != null
          ? StadiumSector.fromJson(json['stadiumSector'])
          : null,
      releasedQuantity: json['releasedQuantity'] is int 
          ? json['releasedQuantity'] 
          : int.tryParse(json['releasedQuantity']?.toString() ?? '0'),
      availableQuantity: json['availableQuantity'] is int 
          ? json['availableQuantity'] 
          : int.tryParse(json['availableQuantity']?.toString() ?? '0'),
      price: json['price'] is double 
          ? json['price'] 
          : double.tryParse(json['price']?.toString() ?? '0.0'),
      usedQuantity: json['usedQuantity'] is int 
          ? json['usedQuantity'] 
          : int.tryParse(json['usedQuantity']?.toString() ?? '0') ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'matchId': matchId,
      'stadiumSector': stadiumSector?.toJson(),
      'releasedQuantity': releasedQuantity,
      'availableQuantity': availableQuantity,
      'price': price,
      'usedQuantity': usedQuantity,
    };
  }
}
